### PR TITLE
Refax of TestEnforcedEncryption to keep on the stable results

### PR DESCRIPTION
**Pull request by ethouris**
The following changes have been applied:

1. Use `srt_epoll_uwait` for epoll-waiting, as this function reports in more comfortable way and doesn't require complicated checking rules. The result is an array of ready sockets together with events on which it is ready.
2. As only the accepted socket is checked for readiness, epoll is subscribed only for listener socket, for IN and ERR. The latter should never be a case, but the subscription remains in order to catch that case should it occur.
3. changed the rules of checking in tests B2-B4. Actually in this test the state on the listener side is a matter of good luck, although the stable rules mandate that it should either result in no connection reporting, or if the connection was reported it should report as broken just after the connecting is finished. Because of that also the report from the epoll waiting function is requested to be ignored, as it may return 0 or 1 depending on luck. Appropriate notes as to why this is working this way was also added.

### CODE REVIEW

File: test/test_enforced_encryption.cpp
Comment (maxsharabayko) [inline test/test_enforced_encryption.cpp:482]: Please disable tracing for the final CI version. Tracing is mainly intended for development stage.
```suggestion
    const bool          m_is_tracing = false;
```

Comment (ethouris) [reply test/test_enforced_encryption.cpp:482]: Damn. Forgot.

File: test/test_enforced_encryption.cpp
Comment (maxsharabayko) [inline test/test_enforced_encryption.cpp:626]: Please leave a blank line at the end of the file

File: test/test_enforced_encryption.cpp
Comment (maxsharabayko) [inline test/test_enforced_encryption.cpp:701]:
```suggestion
CREATE_TEST_CASES(CASE_D_5, Enforced_Off_Off_Pwd_None_None)

```

**Review by maxsharabayko — APPROVED** [review]

**Comment by maxsharabayko** [comment]
Given the tests have passed previously, before the blank line was adding, can be merged without waiting for CI to complete.

**Pull request merged**
